maintenance/wikia/HttpsMigration/migrateCheckUser.php

```diff
@@ -0,0 +1,176 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: abador
+ * Date: 16.04.19
+ * Time: 10:15
+ */
+
+ini_set( 'display_errors', 'stderr' );
+ini_set( 'error_reporting', E_ALL ^ E_NOTICE );
+
+require_once( dirname( __FILE__ ) . '/../../Maintenance.php' );
+
+use \Wikia\Logger\WikiaLogger;
+use Wikia\Util\Statistics\BernoulliTrial;
+
+/**
+ * Script to migrate a user changes log from www.wikia.com to community.fandom.com.
+ *.
+ */
+class MigrateCheckUser extends Maintenance {
+	const BATCH_SIZE = 1000;
+	const CU_LOG_TABLE= 'cu_log';
+	const CU_CHANGES_TABLE= 'cu_changes';
+
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription = 'Migrates user changes logs from www.wikia.com to community.fandom.com';
+		$this->addOption( 'saveChanges', 'Migrate log data for real.',
+			false, false,	'd' );
+	}
+
+	public function execute() {
+		$saveChanges = $this->hasOption( 'saveChanges' );
+
+		$srcDB = wfGetDB( DB_SLAVE, [], 'wikiaglobal' );
+		$dstSlave = wfGetDB( DB_SLAVE, [], 'wikia' );
+		$dstMaster = wfGetDB( DB_MASTER, [], 'wikia' );
+		// this script would generate a ton of db logs, disable them
+		$zeroSampler =  new BernoulliTrial( 0.0 );
+		$srcDB->setSampler( $zeroSampler );
+		$dstSlave->setSampler( $zeroSampler );
+		$dstMaster->setSampler( $zeroSampler );
+
+		$logLastId = 0;
+		$logLastTimestamp = 0;
+		$logDuplicates = 0;
+		$logMigrated = 0;
+		$changesLastId = 0;
+		$changesLastTimestamp = 0;
+		$changesDuplicates = 0;
+		$changesMigrated = 0;
+
+		while(true) {
+			$res = $srcDB->select(
+				self::CU_LOG_TABLE,
+				'*',
+				"cul_id > $logLastId",
+				__METHOD__,
+				[ 'ORDER BY' => 'cul_id ASC', 'LIMIT' => self::BATCH_SIZE ] );
+			$count = $res->numRows();
+			WikiaLogger::instance()->info( 'Processing next batch',
+				[ 'rows' => $count , 'lastTimestamp' => $logLastTimestamp ]
+			);
+			if ($count <= 0) {
+				break;
+			}
+			foreach ( $res as $row ) {
+				$logLastId = $row->cul_id;
+				$logLastTimestamp = $row->cul_timestamp;
+				$duplicateRes = $dstSlave->select(
+					self::CU_LOG_TABLE,
+					'cul_id',
+					[
+						'cul_timestamp' => $row->cul_timestamp,
+						'cul_user' => $row->cul_user,
+						'cul_type' => $row->cul_type
+					],
+					__METHOD__
+				);
```

Comment:
I don't think you can use `cul_timestamp`, `cul_user` and `cul_type` to determine uniqueness/duplication in cu_log:
```
mysql@geo-db-b-slave.query.consul[wikiaglobal]>select count(*)from cu_log where cul_timestamp=20190303174320 and cul_user=4784321 and cul_type='userips';
+----------+
| count(*) |
+----------+
|      112 |
+----------+
1 row in set (0.00 sec)
```